install/0000_30_machine-api-operator_08_webhook.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: machine-api
+  annotations:
+    # create-only so CVO does not trample on the injected cabundle.
+    "service.beta.openshift.io/inject-cabundle": "true"
+    "release.openshift.io/create-only": "true"
+webhooks:
+  - clientConfig:
+      service:
+        name: machine-api-operator-webhook
+        namespace: openshift-machine-api
+        path: /mutate-machine-openshift-io-v1beta1-machine
+    # failurePolicy is ignore so we don't want to block machine lifecycle on the webhook operational aspects.
+    # This would be particularly problematic for chicken egg issues when bootstrapping a cluster.
+    failurePolicy: Ignore
+    matchPolicy: Equivalent
+    name: default.machine.machine.openshift.io
+    rules:
+      - apiGroups:
+          - machine.openshift.io
+        apiVersions:
+          - v1beta1
+        operations:
+          - CREATE
```

Comment:
I was being intentionally conservative to prevent side effects on existing clusters. Post creation if a user tries to remove a defaulted value it should be stopped by validation().
Happy to reconsider if you think it does not make sense.

Reply:
Ok, that makes sense, happy with validation catching it